buildSrc/src/main/java/org/opensearch/gradle/testclusters/StandaloneRestIntegTestTask.java

```diff
@@ -133,6 +134,37 @@ public List<ResourceLock> getSharedResources() {
         return Collections.unmodifiableList(locks);
     }
 
+    private SharedResource getSharedResource(
+        BuildServiceRegistryInternal serviceRegistry,
+        Provider<TestClustersThrottle> throttleProvider
+    ) {
+        try {
+            try {
+                // The forService(Provider) is used by Gradle pre-8.0
```

Comment:
When does this code ever need to run against Gradle pre-8.0 after this change?

Reply:
I will run every time the build will be performed by pre - Gradle 8, best examples - plugins. OpenSearch plugins support is extensively using Gradle plugins, if the plugin uses fe Gradle 7.6 (which at this moment is applicable to any other plugin outside core), the build will fail right away.

Reply:
Gotcha, thanks!